### ISSUE TITLE
Index out of range in peek of TemplateByteScanner

### ISSUE THREAD

**Issue by edmw** [vapor/template-kit#49]
Affected version: 1.1.2

With some of my leaf templates the method `peek` of `TemplateByteScanner` will be called with a negative amount. When the current pointer is less than the absolute amount this will cause a Fatal Error: Index out of range.

Adding a guard `pointer + amount >= 0 else return nil` fixed this for me.

The good thing is, the former version using an unsafe pointer most likely resulted in a buffer overflow.

**Comment by foscomputerservices** [comment]
I added [pull request #48 ](https://github.com/vapor/template-kit/pull/48) to fix this very thing yesterday.  I too encountered this when leaf was scanning the beginning of of file for #//.

```swift
let isEntirelyCommentLine: Bool = (peek(by: -4) == .newLine)
```

**Comment by tanner0101** [comment]
Fixed by #48, thanks!
